pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -383,17 +386,29 @@ func (pc *PodController) configureSecondaryInterface(
 		ifConfigErr = pc.interfaceConfigurator.ConfigureVLANSecondaryInterface(
 			pod.Name, pod.Namespace,
 			podCNIInfo.containerID, podCNIInfo.netNS, network.InterfaceRequest,
-			int(networkConfig.MTU), ipamResult)
+			int(networkConfig.MTU), ipamResult, mac)
 	}
 	return &ipamResult.Result, ifConfigErr
 }
 
 func (pc *PodController) configurePodSecondaryNetwork(pod *corev1.Pod, networkList []*netdefv1.NetworkSelectionElement, podCNIInfo *podCNIInfo) error {
 	usedIFNames := sets.New[string]()
+	usedIFMac := sets.New[string]()
 	for _, network := range networkList {
 		if network.InterfaceRequest != "" {
 			usedIFNames.Insert(network.InterfaceRequest)
 		}
+
+		if network.MacRequest != "" {
```

Comment:
I just realized that we need a `klog.InfoS` message if a MAC request is provided but the network types is SR-IOV.